_posts/2022/2022-10-31-use-zero-for-integer-float-comparison.md

```diff
@@ -0,0 +1,119 @@
+---
+title: "Use zero? for comparison of numerics like Integer, Float and BigDecimal"
+description: "Other languages don't have this"
+layout: article
+category: ruby
+image:
+  base: "2022/use-zero-for-integer-float-comparison"
+  alt: "A zero on a sign on the wall"
+  credit: "Bernard Hermant"
+  source: "https://unsplash.com/photos/rfpSOlH1JlQ"
+---
+
+Compared to other similar languages Ruby often prioritises readability (and joy) when it come to its syntax and methods in its standard library.
+
+An example of this is some syntactic sugar for comparing a value to zero.
+
+See the documentation of the `.zero?` method on [`Integer`](https://ruby-doc.org/core-3.1.2/Integer.html#zero-3F-method), [`Float`](https://ruby-doc.org/core-3.1.2/Float.html#zero-3F-method), [`Numeric`](https://ruby-doc.org/core-3.1.2/Numeric.html#zero-3F-method) and [`BigDecimal`](https://ruby-doc.org/stdlib-3.1.2/libdoc/bigdecimal/rdoc/BigDecimal.html#zero-3F-method).
```

Comment:
*Take a look at the...

Sounds a little more conversational/fun?

Reply:
```suggestion
Take a look at the documentation for the `.zero?` method on [`Integer`](https://ruby-doc.org/core-3.1.2/Integer.html#zero-3F-method), [`Float`](https://ruby-doc.org/core-3.1.2/Float.html#zero-3F-method), [`Numeric`](https://ruby-doc.org/core-3.1.2/Numeric.html#zero-3F-method) and [`BigDecimal`](https://ruby-doc.org/stdlib-3.1.2/libdoc/bigdecimal/rdoc/BigDecimal.html#zero-3F-method).
```